Add Ability to select region for Roborock (#43060)
home-assistant/home-assistant.github.io: source/_integrations/roborock.markdown

{% configuration_basic %}
Email address:
    description: "The email address used to sign in to the Roborock app. A verification code will be sent to this email address when adding the Roborock integration."

> Region:

Reviewer: We discussed offline that we may want to call this the "Roborock Server Region"

>     description: "The region that your Roborock account was created in. Leave this set to Auto unless you are having issues."

Reviewer: ```suggestion
    description: "The region that your Roborock account was created in. Leave this set to Auto unless you are having issues. See troubleshooting steps below."
```

>     description: "The region that your Roborock account was created in. Leave this set to Auto unless you are having issues. See troubleshooting steps below."

Reviewer 2: ```suggestion
    description: "The region that your Roborock account was created in. Leave this set to **Auto** unless you are having issues. See troubleshooting steps below."
```

There is rate limiting built into the Python package that this integration is built on. This is to try to help prevent your instance from overwhelming the Roborock servers and resulting in any kind of IP ban. Best practice is to disable the integration for 24 hours. 

It's also important to try to determine what caused this error in your setup. A common cause some users have is that they have a script that automatically reloads the integration if it goes unavailable. Then, if the device gets stuck and runs out of battery, you are frequently reloading and that causes rate limits.

### When I try to add the integration - it says my region is incorrect

> We recommend using the "Auto" setting for your region, but if that doesn't work because you have accounts in multiple regions here are a few steps we recommend taking:

Reviewer 2: ```suggestion
We recommend using the "Auto" setting for your region. If that doesn't work because you have accounts in multiple regions, try the following steps:
```

> 1) If you ever accidentally created a Roborock account in the wrong region - please make sure you delete it. 

Reviewer: Delete it using the Roborock App?

> 1) If you ever accidentally created a Roborock account in the wrong region - please make sure you delete it. 

Reviewer: Delete it using the Roborock App?

```suggestion
1) If you ever accidentally created a Roborock account in the wrong roborock server region, please make sure you delete it using the Roborock App.
```

> 2) While setting up the integration, you may select the region that you want to sign in with, there are four options: US, EU, RU, CN. It is not always intuitive what region your country is in, so you might have to try a couple. Most countries outside of Russia or China are in US and EU.

Reviewer: Suggesting a slight reword based on our conversation over discord:

```suggestion
2) The Roborock server region for your account may not always be associated with your actual country.  While setting up the integration, you may select the Roborock server region that you want to sign in with, there are four options: US, EU, RU, CN and you may need to try a different region that you would expect. Most users outside of Russia or China are in the US and EU Roborock server regions.
```

> 2. The Roborock server region for your account may not always be associated with your actual country.  While setting up the integration, you may select the Roborock server region that you want to sign in with, with four available options: US, EU, RU, or CN, and you may need to try a different region that you would expect. Most users outside of Russia or China are in the US and EU Roborock server regions.

Reviewer 2: ```suggestion
2. The Roborock server region for your account may not always be associated with your actual country.  While setting up the integration, you may select the Roborock server region that you want to sign in with, with four available options: US, EU, RU, or CN, and you may need to try a different region than the one you expect. Most users outside of Russia or China are in the US and EU Roborock server regions.
```

> 3. Please note that the "Region" that is shown in the app is actually what country your account is registered to and does not always match the region the integration is looking for. 

Reviewer 2: ```suggestion
3. Please note that the "Region" that is shown in the app is actually the country your account is registered to. It does not always match the region the integration is looking for. 
```
